Scripts/player.gd

```diff
@@ -115,7 +115,7 @@ func _physics_process(delta: float) -> void:
 		else:
 			fire_cooldown = FIRE_RATE_MAX  # normal cooldown (0.5)
 		fire_projectile(0.5)
-		print("Active projectiles:", get_active_projectile_count())
+		Logger.log_debug("Active projectiles:", get_active_projectile_count())
```

Comment:
I think this method will interoperate your inputs incorrectly. It's defined as `log_debug(msg: String, class_level_debug_enabled: bool = true)` where the first parameter takes in a string to print out and the second parameter is an optional boolean that allows you to control the debug statements at a class level. That way you don't have to turn off loging for EVERYTHING if you want to shut up a particular object. For `player.gd` we have `
const DEBUG_OBJECT = false` on line 12.

For this line I would use what's called a [format string](https://docs.godotengine.org/en/4.4/tutorials/scripting/gdscript/gdscript_format_string.html) to handle the string input:
```suggestion
		Logger.log_debug("Active projectiles: %s" % get_active_projectile_count(), DEBUG_OBJECT)
```